pyomo/core/plugins/transform/init_vars.py

```diff
@@ -0,0 +1,68 @@
+"""Automatically initialize variables."""
+from __future__ import division
+
+import textwrap
+
+from pyomo.core.base.var import Var
+from pyomo.core.kernel.numvalue import value
+from pyomo.core.plugins.transform.hierarchy import IsomorphicTransformation
+from pyomo.util.plugin import alias
+
+__author__ = "Qi Chen <https://github.com/qtothec>"
+
+
+class InitMidpoint(IsomorphicTransformation):
+    """Initializes variables to the midpoint of their bounds."""
```

Comment:
This documentation does not completely reflect the intent of this transformation.  The checks below consider other cases than just the midpoint computation.